torch/csrc/jit/script/script_type_parser.cpp

```diff
@@ -82,6 +82,13 @@ TypePtr ScriptTypeParser::subscriptToType(
     auto key_type = parseTypeFromExpr(subscript.subscript_exprs()[0]);
     auto value_type = parseTypeFromExpr(subscript.subscript_exprs()[1]);
     return DictType::create(key_type, value_type);
+  } else if (typeName == "Final"){
```

Comment:
This is an incomplete implementation of Final in the parser -- it gives the impression that we support it but then we just ignore it.

Reply:
what should be the proper support here? add Final to the type system?

Reply:
The simplest thing to do would probably be for the constant parser to parse the expect Final before parsing the inner type.

Reply:
do we want to allow Final only when we parse constant or do we want to support it as a general type? we can remove these code if we just want to support it when we parse constant

Reply:
We can only support `Final` as a class constant annotation today, because we don't have const-ness otherwise in the type system. 

I think the best way to structure this is for `ScriptTypeParser::parseConstant` to become `parseClassConstant(Assign assignment)`, which takes care of parsing the `Final` type expr and parsing the rhs value together. This part should be removed to give a correct error message that `Final` doesn't work generically.